zndraw/server/tasks.py

```diff
@@ -256,7 +256,15 @@ def read_file(url: str, target: str, token: str, fileio: dict):
         vis.extend(atoms_list)
         vis.step = len(vis) - 1
     else:
-        vis.upload(target)
+        # resend the current state, present in the database
+        camera = vis.camera
+        step = vis.step
+        vis[:] = list(vis)
```

Comment:
this line is rather slow

Reply:
This is in part slow because `list(vis)` is equivalent to doing `[vis[i] for i in range(len(vis))]` which grabs each frame as a separate transcation. Better would be to pass the slice directly, e.g. `vis[:] = vis[:len(vis)]`.

Reply:
It would be even faster, if we do not gather the entire list at once but do it in steps as well, but I think this is already an improvement over the current version.